api/controllers/FlairController.js

```diff
@@ -132,88 +132,81 @@ module.exports = {
     }
 
     var appData = {
-      limit: 1,
+      limit: 2,
       sort: "createdAt DESC",
       user: req.user.name,
       type: "flairTextChange"
     };
-
-    Event.find(appData).exec(function (err, events) {
-      if (err) {
-        res.status(500).json({error: "Unknown error"});
-      }
-      var now = moment();
-      if (events.length) {
-        var then = moment(events[0].createdAt);
-        then.add(2, 'minutes');
-        if (then.isAfter(now)) {
-          return res.status(400).json({error: "You set your flair too recently, please try again in a few minutes."});
-        }
+    var events = await Event.find(appData);
```

Comment:
According to [the docs](http://sailsjs.org/documentation/reference/waterline-orm/queries#?promises) it supports Bluebird promises, not native promises. Apparently it works with await though.